atomisticparsers/h5md/parser.py

```diff
@@ -0,0 +1,787 @@
+
+                    val = observable.get('value')
+                    if val is not None:
+
+                        value_unit = val.units if hasattr(val, 'units') else None
+                        value_magnitude = val.magnitude if hasattr(val, 'units') else val
+                        if flag_known_property:
+                            property_values_dict['value'] = value_magnitude * value_unit if value_unit else value_magnitude
+                        else:
+                            property_values_dict['value_unit'] = str(value_unit) if value_unit else None
+                            property_values_dict['value_magnitude'] = value_magnitude
+
+                    bins = observable.get('bins')
+                    if bins is not None:
+                        bins_unit = bins.units if hasattr(bins, 'units') else None
+                        bins_magnitude = bins.magnitude if hasattr(bins, 'units') else bins
+                        if flag_known_property:
+                            property_values_dict['bins'] = bins_magnitude * bins_unit if bins_unit else bins_magnitude
+                        else:
+                            property_values_dict['bins_unit'] = str(bins_unit) if bins_unit else None
+                            property_values_dict['bins_magnitude'] = bins_magnitude
+
+                    property_dict[property_value_key].append(property_values_dict)
+                workflow_results[property_key].append(property_dict)
+
+        workflow_results = {}
+        ensemble_average_observables = self.observable_info.get('ensemble_average')
+        ensemble_property_dict = {
+            'property_type_key': 'ensemble_properties',
+            'property_type_value_key': 'ensemble_property_values',
+            'properties_known': {'radial_distribution_functions': 'radial_distribution_function_values'},
+            'property_keys_list': EnsembleProperty.__dict__.keys(),
+            'property_value_keys_list': EnsemblePropertyValues.__dict__.keys()
+        }
+        get_workflow_results(ensemble_property_dict, ensemble_average_observables, workflow_results)
+        correlation_function_observables = self.observable_info.get('correlation_function')
+        correlation_function_dict = {
+            'property_type_key': 'correlation_functions',
+            'property_type_value_key': 'correlation_function_values',
+            'properties_known': {'mean_squared_displacements': 'mean_squared_displacement_values'},
+            'property_keys_list': CorrelationFunction.__dict__.keys(),
+            'property_value_keys_list': CorrelationFunctionValues.__dict__.keys()
+        }
+        get_workflow_results(correlation_function_dict, correlation_function_observables, workflow_results)
+        self.parse_md_workflow(dict(method=workflow_parameters, results=workflow_results))
+
+    def init_parser(self):
+
+        self._data_parser.mainfile = self.filepath
+
+    def parse(self, filepath, archive, logger):
```

Comment:
i suggest we start doing type declarations? we can also do it during the migration.

Reply:
I went through and added declarations. There are still some instances that I am not sure how to deal with, I can go through and point these out to you at the next review time.